src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ArrayConverter.cs

```diff
@@ -3,34 +3,36 @@
 
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Metadata;
 
 namespace System.Text.Json.Serialization.Converters
 {
     /// <summary>
     /// Converter for <cref>System.Array</cref>.
     /// </summary>
-    internal sealed class ArrayConverter<TCollection, TElement> : IEnumerableDefaultConverter<TElement[], TElement>
+    internal sealed class ArrayConverter<TCollection, TElement> : IEnumerableDefaultConverter<TElement[], TElement, List<TElement>>
```

Comment:
I'd caution against adding mode generic parameters to converters, since we'll end up with size regressions in NativeAOT. I think we should leave the accumulator to be implicitly typed for now.

Reply:
I don't think that will increase size since there is 1:1 mapping between intermediate type and collection - here and in other places so it will not expand similarly as casting converter